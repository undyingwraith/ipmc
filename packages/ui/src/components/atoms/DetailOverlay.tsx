import { Backdrop } from '@mui/material';
import { ComponentChildren } from 'preact';

export function DetailOverlay(props: {
	children: ComponentChildren;
}) {
	return (
		<Backdrop
			open={true}
			sx={{
				marginTop: '64px',
				width: '100vw',
				height: 'calc(100vh - 64px)',
			}}
		>
			{props.children}
		</Backdrop>
	);
}
