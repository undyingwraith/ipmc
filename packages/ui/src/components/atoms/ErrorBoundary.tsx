import { Alert } from '@mui/material';
import React, { PropsWithChildren } from 'react';

export class ErrorBoundary extends React.Component<PropsWithChildren<{}>, { hasError: boolean; }> {
	constructor(props: PropsWithChildren<{}>) {
		super(props);
		this.state = { hasError: false };
	}

	static getDerivedStateFromError(error: any) {
		// Update state so the next render will show the fallback UI.
		return { hasError: true };
	}

	componentDidCatch(error: Error, info: any) {
		console.error(error, info);
	}

	render() {
		if (this.state.hasError) {
			return (
				<Alert severity="error">Error</Alert>
			);
		}

		return this.props.children;
	}
}
