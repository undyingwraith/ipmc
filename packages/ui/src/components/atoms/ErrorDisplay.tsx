import { Alert, Typography } from '@mui/material';
import React from 'react';

export function ErrorDisplay(props: { error: Error; }) {
	const { error } = props;

	return (
		<Alert severity={'error'}>
			<Typography data-testid={'error-message'}>{error.name}: {error.message}</Typography>
		</Alert>
	);
}
