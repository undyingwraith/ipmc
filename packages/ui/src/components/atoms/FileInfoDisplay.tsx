import { Box, Stack, Typography } from '@mui/material';
import { IFileInfo, isPosterFeature } from 'ipmc-interfaces';
import React from 'react';
import { useFileUrl } from '../../hooks';
import { useTitle } from '../../hooks/useTitle';

export function FileInfoDisplay(props: { file: IFileInfo; }) {
	const { file } = props;
	const name = useTitle(file);
	const title = (
		<Typography variant={'h3'}>{name}</Typography>
	);
	const posterUrl = useFileUrl(isPosterFeature(file) && file.posters.length > 0 ? file.posters[0]?.cid : undefined);

	return isPosterFeature(file) && file.posters.length > 0 ? (
		<Stack direction={'row'} spacing={1}>
			<img src={posterUrl.value} style={{ height: 250, flexGrow: 0 }} />
			{title}
		</Stack>
	) : (
		<Box>
			{title}
		</Box>
	);
}
