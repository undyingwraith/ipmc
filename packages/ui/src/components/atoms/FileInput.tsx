import { Box, Button } from '@mui/material';
import { ReadonlySignal, Signal, useComputed } from '@preact/signals';
import { useTranslation } from '../../hooks';

interface IFileInputProps {
	label?: ReadonlySignal<string>;
	value: Signal<File[]>;
	accept?: string;
}

export function FileInput(props: IFileInputProps) {
	const _t = useTranslation();

	return (
		<Box>
			<Button
				variant={'contained'}
				component={'label'}
			>
				{props.label?.value ?? _t('UploadFile')}
				<input
					type={'file'}
					hidden
					accept={props.accept}
					onChange={(ev) => {
						props.value.value = Array.from(ev.currentTarget.files!);
					}}
				/>
			</Button>
			{useComputed(() => props.value.value.map((f) => (
				<Box>
					{f.name}
				</Box>
			)))}
		</Box>
	);
}
