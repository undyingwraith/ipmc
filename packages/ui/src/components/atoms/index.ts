export { ErrorBoundary } from './ErrorBoundary';
export { FileInput } from './FileInput';
export { FormList } from './FormList';
export { Identicon } from './Identicon';
export { ImageView } from './ImageView';
export { Loader } from './Loader';
export { SelectInput } from './SelectInput';
export { Spacer } from './Spacer';
export { TextInput } from './TextInput';
export { ThemeToggle } from './ThemeToggle';
