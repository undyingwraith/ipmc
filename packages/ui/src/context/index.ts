export { AppContextProvider, useApp, useService, useOptionalService } from './AppContext';
export { type IThemeContext, ThemeContextProvider, useTheme } from './ThemeContext';
