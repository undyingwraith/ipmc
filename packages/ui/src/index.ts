export { IpmcApp } from './components/pages/IpmcApp';
