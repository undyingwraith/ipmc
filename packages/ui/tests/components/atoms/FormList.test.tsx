import { Signal } from '@preact/signals';
import { render } from '@testing-library/preact';
import { CoreModule } from 'ipmc-core';
import React from 'react';
import { FormList } from 'src/components/atoms';
import { AppContextProvider } from 'src/context';
import { describe, expect, test } from 'vitest';

describe('FormList', () => {
	test('Add button works', () => {
		const values = new Signal<Signal<{ name: string; }>[]>([]);

		const t = render(
			<AppContextProvider setup={(app) => {
				app.use(CoreModule);
			}}>
				<FormList
					createItem={() => ({ name: 'test' })}
					renderControl={(data) => <div>{data.value.name}</div>}
					label={new Signal('Test')}
					values={values}
				/>
			</AppContextProvider>
		);
		t.findByText('<Add>').then(c => c.click());

		expect(t.container).toMatchSnapshot();
	});
});
