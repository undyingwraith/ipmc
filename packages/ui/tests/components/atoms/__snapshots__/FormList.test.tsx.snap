// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormList > Add button works 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-1 css-octady-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-8 css-1cdctrw-MuiGrid2-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        Test
      </p>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 css-17yh78r-MuiGrid2-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-uf9a8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        &lt;Add&gt;
      </button>
    </div>
  </div>
</div>
`;
